File: core/connectors/mysql_connector_engine.py
from __future__ import annotations

from typing import Any, Dict, Optional


def extract_mysql_runtime(
    snapshot: Optional[Dict[str, Any]] = None,
) -> Dict[str, Any]:
    snap = snapshot or {}
    return {
        "database_type": "mysql",
        "schemas": list(snap.get("schemas", [])),
        "tables": sorted(snap.get("tables", []), key=str),
        "indexes": list(snap.get("indexes", [])),
        "metrics": dict(snap.get("metrics", {})),
        "active_connections": int(snap.get("active_connections", 0)),
        "replication_state": str(snap.get("replication_state", "")),
        "degraded": snap.get("degraded", False),
        "bounded": True,
    }

File: src/connectors/mysqlConnectorEngine.ts
/**
 * Converted from Python: core/connectors/mysql_connector_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function extractMysqlRuntime(snapshot: any = null): any {
  var snap: any = py.or2(snapshot, () => ({}));
  return {"database_type": "mysql", "schemas": [...py.iter(py.get(snap, "schemas", []))], "tables": py.sorted(py.get(snap, "tables", []), {key: (py.toStr) as (item: any) => any}), "indexes": [...py.iter(py.get(snap, "indexes", []))], "metrics": py.pyDict(py.get(snap, "metrics", {})), "active_connections": py.toInt(py.get(snap, "active_connections", 0)), "replication_state": py.toStr(py.get(snap, "replication_state", "")), "degraded": py.get(snap, "degraded", false), "bounded": true};
}
